src/webview/html/visjsDebuggerPanel.html

```diff
@@ -24,38 +25,110 @@
-          edges.updateOnly(edges.map((edge) => ({ ...edge, color: defaultEdgeColor })));
-
-          nodes.add(message.data.addNodes);
-          nodes.updateOnly(message.data.updateNodes);
-          edges.add(message.data.addEdges);
-        } else if (message.command === 'exportVisjs') {
-          const canvas = document.querySelector('#visjs-network canvas');
-          const img = canvas.toDataURL('image/png');
-          var downloadLink = document.createElement('a');
-          downloadLink.download = 'export.png';
-          downloadLink.href = img;
-          downloadLink.click();
+        switch (message.command) {
+          case 'initializeVisjs':
+            const container = document.querySelector('#visjs-network');
+            edges = new vis.DataSet(message.data.edges);
+            nodes = new vis.DataSet(message.data.nodes);
+            defaultNodeColor = message.options.nodes.color;
+            defaultEdgeColor = message.options.edges.color;
+            const data = {
+              edges,
+              nodes,
+            };
+            network = new vis.Network(container, data, message.options);
+            vscode = acquireVsCodeApi();
+            canvas = document.querySelector('#visjs-network canvas');
+            break;
+          case 'updateVisjs':
+            edges.remove(message.data.deleteEdgeIds);
+            nodes.remove(message.data.deleteNodeIds);
+            nodes.updateOnly(
+              nodes.map((node) => ({
+                ...node,
+                color: defaultNodeColor,
+              }))
+            );
+            edges.updateOnly(
+              edges.map((edge) => ({
+                ...edge,
+                color: defaultEdgeColor,
+              }))
+            );
+            nodes.add(message.data.addNodes);
+            nodes.updateOnly(message.data.updateNodes);
+            edges.add(message.data.addEdges);
+            break;
+          case 'exportVisjs':
+            const img = canvas.toDataURL('image/png');
+            downloadFile('export.png', img);
+            break;
+          case 'startRecordingVisjs':
+            chunks = [];
+            const stream = canvas.captureStream();
+            recorder = new MediaRecorder(stream, {
+              mimeType: 'video/webm;codecs=vp9',
+              ignoreMutedMedia: true,
+            });
+            recorder.ondataavailable = (e) => {
+              if (e.data.size > 0) {
+                chunks.push(e.data);
+              }
+            };
+            recorder.start();
+            break;
+          case 'stopRecordingVisjs':
+            recorder.stop();
+            vscode.postMessage('creatingGif');
+            setTimeout(() => {
+              const blob = new Blob(chunks, {
+                type: 'video/webm',
+              });
+              const webmVideo = URL.createObjectURL(blob);
+              convertWebmToGif(webmVideo);
```

Comment:
If we need to record a video to make a GIF image, we might as well skip the conversion and make the video available for download.
On the other hand, the GIF image may be better suited for post-processing (e.g. omitting unwanted frames).
Should we offer both a GIF and a WEBM download?

Reply:
The Idea behind a Gif export was that GIF is more widely accepted by third party programms. For example, it could be easier integrated in PowerPoint. But WEBM would have the benefits that we could create it much faster. We wouldn’t need to wait for the conversion to a GIF. I think that the option to export directly as WEBM would be useful especially when the user wants to export a longer recording. I will create a new ticket in Jira where we can discuss this feature further.